### PR TITLE
.Net: Make ONNX connector Native-AOT compatible.

### PR DESCRIPTION
**Pull request by SergeyMenshykh**
### Motivation and Context
This PR makes the ONNX connector Native-AOT friendly, allowing it to be used in applications targeting Native-AOT.

### Description
- Adds a new parameter of `JsonSerializerOptions` type to the constructor of the `OnnxRuntimeGenAIChatCompletionService` class and a few extension methods.  
- Adds a new `OnnxRuntimeGenAIPromptExecutionSettings.FromExecutionSettings` method that accepts JSOs to serialize prompt execution settings type of which is not known in advance.  
- Makes function choice behavior classes public, allowing the source generator to access them and not fail with the error: "_dotnet\src\Connectors\Connectors.Onnx\Text\OnnxRuntimeGenAIPromptExecutionSettingsJsonSerializerContext - Copy.cs(10,31,10,91): warning SYSLIB1222: The constructor on type 'Microsoft.SemanticKernel.AutoFunctionChoiceBehavior' has been annotated with JsonConstructorAttribute but is not accessible by the source generator. (https://learn.microsoft.com/dotnet/fundamentals/syslib-diagnostics/syslib1222)_"

CC: @eiriktsarpalis

### CODE REVIEW

File: dotnet/src/Connectors/Connectors.Onnx/Text/OnnxRuntimeGenAIPromptExecutionSettingsJsonSerializerContext - Copy.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Microsoft.SemanticKernel.Connectors.Onnx;
+
+namespace Microsoft.SemanticKernel.Text;
+
+[JsonSerializable(typeof(OnnxRuntimeGenAIPromptExecutionSettings))]
+internal sealed partial class OnnxRuntimeGenAIPromptExecutionSettingsJsonSerializerContext : JsonSerializerContext
```
Comment (westey-m) [inline dotnet/src/Connectors/Connectors.Onnx/Text/OnnxRuntimeGenAIPromptExecutionSettingsJsonSerializerContext - Copy.cs:10]: The filename ends in `- Copy.cs`, which I assume is inadvertent.

File: dotnet/src/SemanticKernel.Abstractions/AI/FunctionChoiceBehaviors/AutoFunctionChoiceBehavior.cs [hunk 2]
```diff
@@ -10,7 +11,8 @@ namespace Microsoft.SemanticKernel;
 /// Represents a <see cref="FunctionChoiceBehavior"/> that provides either all of the <see cref="Kernel"/>'s plugins' functions to AI model to call or specified ones.
 /// This behavior allows the model to decide whether to call the functions and, if so, which ones to call.
 /// </summary>
-internal sealed class AutoFunctionChoiceBehavior : FunctionChoiceBehavior
+[Experimental("SKEXP0001")]
```
Comment (westey-m) [inline dotnet/src/SemanticKernel.Abstractions/AI/FunctionChoiceBehaviors/AutoFunctionChoiceBehavior.cs:14]: Is this related to the AOT change?

Comment (westey-m) [inline dotnet/src/SemanticKernel.Abstractions/AI/FunctionChoiceBehaviors/AutoFunctionChoiceBehavior.cs:14]: Consider making the constructors internal, so that even though the class is public, we don't have to worry about keeping the constructors backward compatible.

**Review by westey-m — APPROVED** [review]

**Review by markwallace-microsoft — APPROVED** [review]

File: dotnet/src/SemanticKernel.Abstractions/AI/FunctionChoiceBehaviors/AutoFunctionChoiceBehavior.cs (hunk 2 above)
Comment (SergeyMenshykh) [reply dotnet/src/SemanticKernel.Abstractions/AI/FunctionChoiceBehaviors/AutoFunctionChoiceBehavior.cs:14]: I tried that and ended up with reflection-based code to help the YamlDotNet serializer instantiate those classes, as it doesn't support non-public constructors out of the box. Then I thought we might eventually end up making those constructors public anyway, because for Yaml deserialization in Native-AOT, we might need to use a source code generation equivalent. Let me try to make the change, and let's discuss it tomorrow.

File: dotnet/src/Connectors/Connectors.Onnx/Text/OnnxRuntimeGenAIPromptExecutionSettingsJsonSerializerContext - Copy.cs (hunk 1 above)
Comment (SergeyMenshykh) [reply dotnet/src/Connectors/Connectors.Onnx/Text/OnnxRuntimeGenAIPromptExecutionSettingsJsonSerializerContext - Copy.cs]: Oops, the file name is log so I did not noticed the suffix :) Fixed.

File: dotnet/src/Connectors/Connectors.Onnx/OnnxRuntimeGenAIPromptExecutionSettings.cs [hunk 3]
```diff
@@ -28,10 +32,34 @@ public static OnnxRuntimeGenAIPromptExecutionSettings FromExecutionSettings(Prom
             return settings;
         }
 
-        var json = JsonSerializer.Serialize(executionSettings);
+        var json = JsonSerializer.Serialize<object>(executionSettings);
```
Comment (eiriktsarpalis) [inline dotnet/src/Connectors/Connectors.Onnx/OnnxRuntimeGenAIPromptExecutionSettings.cs:35]: Why is this being serialized as object? Seems like the type is known and you might have been able to avoid the annotation in the method.
